src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,50 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<F,S> {
+    private F first;
+    private S second;
 
+    private Pair(F first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public F getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    public static <F, S> Pair of(F first, S second) {
+        return new Pair(first, second);
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == null) {
+            return false;
+        }
+        if (object.getClass().equals(Pair.class)) {
+            if (this == object) {
+                return true;
+            }
+            Pair pair = (Pair) object;
```

Comment:
here as well - let's use parameterized Pair - not raw

Reply:
done